pkg/azure/utils.go

```diff
@@ -67,20 +75,133 @@ func SetupAzureTesting(subscriptionId string, testName string) string {
+				Filter: proto.String(fmt.Sprintf("tagName eq '%s' and tagValue eq '%s'", namespaceTagKey, namespace)),
+			})
+			resourceTypeToResources := make(map[string][]*armresources.GenericResourceExpanded)
+			resourceTypeToAPIVersion := make(map[string]string)
+			for pager.More() {
+				result, err := pager.NextPage(ctx)
+				if err != nil {
+					panic(fmt.Sprintf("failed to get next page of resources: %v", err))
+				}
+
+				for _, resource := range result.Value {
+					resourceIDInfo, err := getResourceIDInfo(*resource.ID)
+					if err != nil {
+						panic(fmt.Sprintf("Unable to parse resource ID info: %v", err))
+					}
+
+					// Need to check here since resourceGroup can't be used with tags in the above filter
+					// Ignore case when checking because Azure sometimes capitalizes the resource group name for no apparent reason
+
+					if strings.EqualFold(resourceIDInfo.ResourceGroupName, resourceGroupName) {
+						if _, ok := resourceTypeToResources[*resource.Type]; !ok {
+							// Initialize list
+							resourceTypeToResources[*resource.Type] = make([]*armresources.GenericResourceExpanded, 0)
+							// Find correct API version
+							_, ok := resourceTypeToAPIVersion[*resource.Type]
+							if !ok {
+								providerNamespace := strings.Split(*resource.Type, "/")[0]
+								resp, err := providersClient.Get(ctx, providerNamespace, nil)
+								if err != nil {
+									panic(fmt.Errorf("Unable to get provider resource type: %w", err))
+								}
+								// Store all API versions under this provider namespace to reduce potential duplicate requests
+								for _, resourceType := range resp.Provider.ResourceTypes {
+									// Breakdown of the term "resource type" overloading
+									// - *resource.Type = "Microsoft.Compute/virtualMachines"
+									// - providerNamespace = "Microsoft.Compute"
+									// - *resourceType.ResourceType is one of "virtualMachines", "disks", etc.
+									resourceTypeToAPIVersion[providerNamespace+"/"+*resourceType.ResourceType] = *resourceType.APIVersions[0] // Use most recent API version
+								}
+							}
+						}
+						resourceTypeToResources[*resource.Type] = append(resourceTypeToResources[*resource.Type], resource)
+					}
+
+				}
+				// Delete resources in the following order
+				// TODO @seankimkdy: figure out where clusters fit in on this order
+				deletionOrder := []string{
+					virtualMachineTypeName,
+					networkInterfaceTypeName,
+					diskTypeName,
+					connectionTypeName,
+					virtualNetworkGatewayTypeName,
+					localNetworkGatewayTypeName,
+					publicIPAddressTypeName,
+					virtualNetworkTypeName,
+					networkSecurityGroupTypeName,
+					networkWatcherTypeName,
+				}
+				for _, resourceType := range deletionOrder {
+					if resources, ok := resourceTypeToResources[resourceType]; ok {
+						err = deleteResources(ctx, resourcesClient, resources, resourceTypeToAPIVersion[resourceType])
+						if err != nil {
```

Comment:
We'll have to see if this is necessary (especially since this is to help with manual testing anyways), but if the failure rate for deletions is high enough, it might be worth having a retry mechanism here

Reply:
Yea I'll keep that in the back of my mind. We can cross that bridge if more people use this option and it becomes an issue.